sw/device/tests/clkmgr_external_clk_src_for_sw_test.c

```diff
@@ -0,0 +1,143 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// This test enables the external clock via software and ramps down the
+// clock dividers. It checks the expected frequencies via the clock count
+// measurement feature.
+//
+// The variability for USB is larger because that clock is uncalibrated at
+// power-on. We can tighten up the USB measurement once
+// https://github.com/lowRISC/opentitan/issues/11264 is addressed.
```

Comment:
nit: maybe add a `// TODO(lowRISC/opentitan:#11264): tighten up the USB measurement once this issue is addressed`?